Libraries/src/net/relinc/libraries/application/LineChartWithMarkers.java

```diff
@@ -6,182 +6,421 @@
-        }
-
-        public void addHorizontalValueMarker(Data<X, Y> marker) {
-            Objects.requireNonNull(marker, "the marker must not be null");
-            if (horizontalMarkers.contains(marker)) return;
-            Line line = new Line();
-            marker.setNode(line );
-            getPlotChildren().add(line);
-            horizontalMarkers.add(marker);
-        }
-
-        public void removeHorizontalValueMarker(Data<X, Y> marker) {
-            Objects.requireNonNull(marker, "the marker must not be null");
-            if (marker.getNode() != null) {
-                getPlotChildren().remove(marker.getNode());
-                marker.setNode(null);
-            }
-            horizontalMarkers.remove(marker);
-        }
-
-        public void addVerticalValueMarker(Data<X, Y> marker) {
-            Objects.requireNonNull(marker, "the marker must not be null");
-            if (verticalMarkers.contains(marker)) return;
-            Line line = new Line();
-            line.setFill(Color.GRAY);
-            line.setStroke(Color.GRAY);
-            marker.setNode(line );
-            getPlotChildren().add(line);
-            verticalMarkers.add(marker);
-            setMouseTransparentToEverythingButBackground();
-        }
-        public void addVerticalValueMarker(Data<X, Y> marker, Color col) {
-            Objects.requireNonNull(marker, "the marker must not be null");
-            if (verticalMarkers.contains(marker)) return;
-            Line line = new Line();
-            line.setFill(col);
-            line.setStroke(col);
-            line.setStrokeWidth(2);
-            marker.setNode(line );
-            getPlotChildren().add(line);
-            verticalMarkers.add(marker);
-            setMouseTransparentToEverythingButBackground();
-        }
-
-        public void removeVerticalValueMarker(Data<X, Y> marker) {
-            Objects.requireNonNull(marker, "the marker must not be null");
-            if (marker.getNode() != null) {
-                getPlotChildren().remove(marker.getNode());
-                marker.setNode(null);
-            }
-            verticalMarkers.remove(marker);
-            
-        }
-        public void clearVerticalMarkers(){
-        	for(Data<X,Y> d : verticalMarkers){
-        		getPlotChildren().remove(d.getNode());
-        	}
-        	for(Data<X,X> d : verticalRangeMarkers){
-        		getPlotChildren().remove(d.getNode());
-        	}
-        	
-        	verticalMarkers.clear();
-        	verticalRangeMarkers.clear();
-        }
-        public void clearHorizontalMarkers(){
-        	for(Data<X,Y> d : horizontalMarkers){
-        		getPlotChildren().remove(d.getNode());
-        	}
+
+    public chartDataType xDataType;
+    public chartDataType yDataType;
+    public chartType thisChartType;
+
+    public enum chartType {
+        STRESSSTRAIN, STRESSTIME, STRAINTIME, STRAINRATETIME, FACEFORCETIME, LOADDISPLACEMENT, LOADTIME, DISCPLACEMENTTIME, DISPLACEMENTRATETIME;
+    }
```

Comment:
I don't think this enum is necessary, can't we just use `xDataType` and `yDataType` to identify our chart types? It seems like redundant information